interfaces/acados_matlab_octave/AcadosSim.m

```diff
@@ -156,9 +156,150 @@ function make_consistent(self)
                     error(['ERK: num_stages = ', num2str(self.solver_options.num_stages) ' not available. Only number of stages = {1,2,3,4} implemented!']);
                 end
             end
+        end
+
+        function generate_external_functions(self)
+            if nargin < 2
+                % options for code generation
+                code_gen_opts = struct();
+                code_gen_opts.generate_hess = self.solver_options.sens_hess;
+                code_gen_opts.code_export_directory = 'c_generated_code'; % TODO: for OCP this is part of OCP class
+                code_gen_opts.ext_fun_expand_dyn = self.solver_options.ext_fun_expand_dyn;
+                code_gen_opts.ext_fun_expand_cost = false;
+                code_gen_opts.ext_fun_expand_constr = false;
+                code_gen_opts.ext_fun_expand_precompute = false;
+
+                context = GenerateContext(self.model.p_global, self.model.name, code_gen_opts);
+            else
+                code_gen_opts = context.code_gen_opts;
+            end
+
+            model_dir = fullfile(pwd, code_gen_opts.code_export_directory, [self.model.name '_model']);
+            check_dir_and_create(model_dir);
+
+            if strcmp(self.model.dyn_ext_fun_type, 'generic')
+                copyfile(fullfile(pwd, self.model.dyn_generic_source), model_dir);
+                context.add_external_function_file(ocp.model.dyn_generic_source, model_dir);
+
+            elseif strcmp(self.model.dyn_ext_fun_type, 'casadi')
+                import casadi.*
+                check_casadi_version();
+                switch self.solver_options.integrator_type
+                    case 'ERK'
+                        generate_c_code_explicit_ode(context, self.model, model_dir);
+                    case 'IRK'
+                        generate_c_code_implicit_ode(context, self.model, model_dir);
+                    case 'GNSF'
+                        generate_c_code_gnsf(context, self.model, model_dir);
+                    case 'DISCRETE'
```

Comment:
does this actually work? I don't think we have an acados integrator for discrete dynamics